airbyte-cdk/python/airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py

```diff
@@ -232,7 +238,9 @@ def create_add_fields(self, model: AddFieldsModel, config: Config, **kwargs) ->
         return AddFields(fields=added_field_definitions, parameters=model.parameters)
 
     @staticmethod
-    def create_api_key_authenticator(model: ApiKeyAuthenticatorModel, config: Config, **kwargs) -> ApiKeyAuthenticator:
+    def create_api_key_authenticator(
```

Comment:
I think we should update the interface to pass in a Union[SessionTokenAuthenticatorModel, TokenProvider]

With the current implementation, it's not obvious that the model is ignored if a token provider is set. We could add a check to verify that the model.api_token is `""`,  but using a `Union` makes it clear that we only expect one or the other

Reply:
The model contains more than just api_key, that's the only bit that is ignored, so the union wouldn't work. Do you still think it makes sense?

Reply:
can we fail if the API key is set?

Reply:
Sounds like a good approach